mender.go

```diff
@@ -315,7 +315,7 @@ func (m *mender) doBootstrap() menderError {
 }
 
 func (m *mender) FetchUpdate(url string) (io.ReadCloser, int64, error) {
-	return m.updater.FetchUpdate(m.api.Request(m.authToken), url)
+	return m.updater.FetchUpdate(m.api, url)
```

Comment:
I'm wondering how to distinguish whether we need to pass authorization and if so, what kind of authorization. 

For instance, if download URL has the same prefix (or at least host/port/scheme) as server/backend URL would it be ok to assume that we should use API token?